### PR TITLE
Bugfix: wrong attention mask calculation resulted in wrong embeddings

### PR DESCRIPTION
**Pull request by maziyarpanahi**
This pull request includes several changes to the `BGEEmbeddings` class in `python/sparknlp/annotator/embeddings/bge_embeddings.py` and various Scala files to add new functionality and improve code quality. The most important changes include adding support for the CLS token in sentence embeddings, updating the default pretrained model, and various code refactorings to improve readability and maintainability.

### Enhancements to `BGEEmbeddings`:

* Added `HasClsTokenProperties` to `BGEEmbeddings` to support the use of the CLS token for sentence embeddings. (`python/sparknlp/annotator/embeddings/bge_embeddings.py`) [[1]](diffhunk://#diff-6ad6d32ff35cbca9c6dafea6a02f2b595e7459d7ffe807535f645b8df34e9400L24-R25) [[2]](diffhunk://#diff-6ad6d32ff35cbca9c6dafea6a02f2b595e7459d7ffe807535f645b8df34e9400R64-R65) [[3]](diffhunk://#diff-6ad6d32ff35cbca9c6dafea6a02f2b595e7459d7ffe807535f645b8df34e9400R154)
* Updated the default pretrained model name to `bge_small_en_v1.5`. (`python/sparknlp/annotator/embeddings/bge_embeddings.py`)

### New Class and Methods:

* Introduced `HasClsTokenProperties` class to handle CLS token properties. (`python/sparknlp/common/properties.py`)

### Refactorings and Improvements:

* Refactored `BGE` class to include `useCLSToken` parameter in various methods and improved tensor handling. (`src/main/scala/com/johnsnowlabs/ml/ai/BGE.scala`) [[1]](diffhunk://#diff-8ae520dba7a468f556fa87352696cf83b7cd41d3a287add8792cf59b293aa4c0L20-R20) [[2]](diffhunk://#diff-8ae520dba7a468f556fa87352696cf83b7cd41d3a287add8792cf59b293aa4c0L74-R81) [[3]](diffhunk://#diff-8ae520dba7a468f556fa87352696cf83b7cd41d3a287add8792cf59b293aa4c0L171-R183) [[4]](diffhunk://#diff-8ae520dba7a468f556fa87352696cf83b7cd41d3a287add8792cf59b293aa4c0L218-R221) [[5]](diffhunk://#diff-8ae520dba7a468f556fa87352696cf83b7cd41d3a287add8792cf59b293aa4c0R238-L261) [[6]](diffhunk://#diff-8ae520dba7a468f556fa87352696cf83b7cd41d3a287add8792cf59b293aa4c0L283-R298) [[7]](diffhunk://#diff-8ae520dba7a468f556fa87352696cf83b7cd41d3a287add8792cf59b293aa4c0L297-R312)
* Removed unnecessary blank lines and improved code formatting in `Albert` and `Bart` classes. (`src/main/scala/com/johnsnowlabs/ml/ai/Albert.scala`, `src/main/scala/com/johnsnowlabs/ml/ai/Bart.scala`) [[1]](diffhunk://#diff-f47bb266a6298a48cec2659dc5abbd7ae15478f166bcd596d0997be7ab410629L163-L167) [[2]](diffhunk://#diff-f47bb266a6298a48cec2659dc5abbd7ae15478f166bcd596d0997be7ab410629L195-L196) [[3]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L24-R26) [[4]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L78) [[5]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L105) [[6]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L240) [[7]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L301-R304) [[8]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L363-R389) [[9]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L407-L408) [[10]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L438-L446) [[11]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L457-R474) [[12]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L482-L486) [[13]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L513) [[14]](diffhunk://#diff-24d7c95ef288ae2a780bd4cb452d7b0c82aa29ac7ed8324d8d6d5a3913cdd692L709-R726)

These changes collectively enhance the functionality of the `BGEEmbeddings` class and improve the overall code quality in related files.This pull request includes several updates to the attention mask logic across multiple classes in the `com.johnsnowlabs.ml.ai` package. The changes primarily involve modifying the condition used to set the attention mask values.

### Updates to attention mask logic:

* [`BGE.scala`](diffhunk://#diff-8ae520dba7a468f556fa87352696cf83b7cd41d3a287add8792cf59b293aa4c0L171-R181): Changed the condition for setting attention mask values from `x < 0L` to `x == 0L` in both `getSentenceEmbeddingFromOv` and `getSentenceEmbeddingFromOnnx` methods. [[1]](diffhunk://#diff-8ae520dba7a468f556fa87352696cf83b7cd41d3a287add8792cf59b293aa4c0L171-R181) [[2]](diffhunk://#diff-8ae520dba7a468f556fa87352696cf83b7cd41d3a287add8792cf59b293aa4c0L218-R218)
* [`E5.scala`](diffhunk://#diff-c3258f1c36fd9c7b4ad970c55763db2845fb8df58541f9523c8570ac3fd31c23L172-R172): Updated the attention mask condition from `x < 0L` to `x == 0L` in the `getSentenceEmbeddingFromOnnx` method. [[1]](diffhunk://#diff-c3258f1c36fd9c7b4ad970c55763db2845fb8df58541f9523c8570ac3fd31c23L172-R172) [[2]](diffhunk://#diff-c3258f1c36fd9c7b4ad970c55763db2845fb8df58541f9523c8570ac3fd31c23L221-R221)
* [`MPNet.scala`](diffhunk://#diff-777c70478f1c85fa42f6a697466264c0fd5ec4eb5a950e9eed488177486c77b2L185-R185): Modified the attention mask condition from `x < this.paddingTokenId` to `x == this.paddingTokenId` in the `getSentenceEmbeddingFromOv` and `getSentenceEmbeddingFromOnnx` methods. [[1]](diffhunk://#diff-777c70478f1c85fa42f6a697466264c0fd5ec4eb5a950e9eed488177486c77b2L185-R185) [[2]](diffhunk://#diff-777c70478f1c85fa42f6a697466264c0fd5ec4eb5a950e9eed488177486c77b2L220-R220)
* [`Mxbai.scala`](diffhunk://#diff-cd44bd93f8a27d61da8f3dd50e696192d0c0add7398fa465e135067e287fbdacL217-R217): Changed the attention mask condition from `x < 0L` to `x == 0L` in the `getSentenceEmbeddingFromOnnx` method.
* [`Nomic.scala`](diffhunk://#diff-bfd0602246f63d27abf9073eaf8991192132120f7a74286f4c30ebc6241e1dbcL77-R77): Updated the attention mask condition from `x < 0L` to `x == 0L` in the `getSentenceEmbeddingFromOnnx` method. [[1]](diffhunk://#diff-bfd0602246f63d27abf9073eaf8991192132120f7a74286f4c30ebc6241e1dbcL77-R77) [[2]](diffhunk://#diff-bfd0602246f63d27abf9073eaf8991192132120f7a74286f4c30ebc6241e1dbcL117-R117)
* [`SnowFlake.scala`](diffhunk://#diff-051737ec8f1453b2916b455949cd678e57a97e0c35d3a5bc939f732b57e10651L226-L248): Modified the attention mask condition from `x < 0L` to `x == 0L` in both `getSentenceEmbeddingFromOv` and `getSentenceEmbeddingFromOnnx` methods. [[1]](diffhunk://#diff-051737ec8f1453b2916b455949cd678e57a97e0c35d3a5bc939f732b57e10651L226-L248) [[2]](diffhunk://#diff-051737ec8f1453b2916b455949cd678e57a97e0c35d3a5bc939f732b57e10651L272-R263)
* [`UAE.scala`](diffhunk://#diff-bae222bce5a03fa1abb68ee56bed18d5d6f3e4bc90c5d954877a51f0183f2220L216): Changed the attention mask condition from `x < 0L` to `x == 0L` in both `getSentenceEmbeddingFromOpenvino` and `getSentenceEmbeddingFromOnnx` methods. [[1]](diffhunk://#diff-bae222bce5a03fa1abb68ee56bed18d5d6f3e4bc90c5d954877a51f0183f2220L216) [[2]](diffhunk://#diff-bae222bce5a03fa1abb68ee56bed18d5d6f3e4bc90c5d954877a51f0183f2220L227-L249) [[3]](diffhunk://#diff-bae222bce5a03fa1abb68ee56bed18d5d6f3e4bc90c5d954877a51f0183f2220L273-R263)

### CODE REVIEW

**Review by ahmedlone127 — COMMENTED** [review]
LGTM!
